src/app/shared/services/dynamic-data/dynamic-data.service.ts

```diff
@@ -66,9 +66,41 @@ export class DynamicDataService extends AsyncServiceBase {
   }
   private registerTemplateActionHandlers() {
     this.templateActionRegistry.register({
+      /**
+       * Write properties on the current item (default), or on an explicitly targeted item,
+       * e.g.
+       * click | set_item | completed:true;
+       * click | set_item | _id: @item.id, completed:true;
+       * click | set_item | _index: @item._index + 1, completed:true;
+       */
       set_item: async ({ args, params }) => {
-        const [flow_name, row_id] = args;
-        await this.update("data_list", flow_name, row_id, params);
+        // data-items component populates data_list name, along with an array of all item ids
+        // and the index of the current item triggering the action
+        const [flow_name, itemDataIDs, itemIndex] = args;
+        const { _index, _id, ...writeableProps } = params;
+
+        // Target current row if another target is not explicitly provided
+        let targetRowId = itemDataIDs[itemIndex];
+        if (_index !== undefined) {
+          targetRowId = itemDataIDs[_index];
+        }
+        if (_id) {
+          targetRowId = _id;
+        }
+
+        if (itemDataIDs.includes(targetRowId)) {
+          console.log(
+            `[SET ITEM] - Setting properties on ${targetRowId}: ${JSON.stringify(writeableProps)}`
+          );
+          await this.update("data_list", flow_name, targetRowId, writeableProps);
+        } else {
+          if (_id) {
+            console.warn(`[SET ITEM] - No item with ID ${_id}`);
+          }
+          if (_index !== undefined) {
+            console.warn(`[SET ITEM] - No item at index ${_index}`);
+          }
```

Comment:
nit(blocking): I feel like this is a large amount of item-specific logic being handled in the dynamic-data service. Would it be possible to move the item selection logic to the items component instead, so that the action is always triggered with the correct item index instead?
There's also quite some debug logs that I think probably should make it into production (would suggest remove console log and merge warn statements